src/back/glsl/mod.rs

```diff
@@ -368,6 +404,7 @@ impl<'a, W: Write> Writer<'a, W> {
 
             block_id: IdGenerator::default(),
             named_expressions: crate::NamedExpressions::default(),
+            unused_unsized_types: unused_unsized_structs,
```

Comment:
in fact, even this array isn't needed.
The current logic of the backend is such that the struct is only written down together with the global variable itself. This logic needs to be extended, but for your fix, the *only* extra logic is needed is checking that `ep_info` before calling `write_global`.

Reply:
I also need the list of unused structs in the `collect_required_features()`, though.

Reply:
We could actually remove the field and then make it an argument of `collect_required_features()` though if you thought that was better.